benchmark/url/url-searchparams-iteration.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const common = require('../common.js');
+const URLSearchParams = require('url').URLSearchParams;
+const v8 = require('v8');
+
+v8.setFlagsFromString('--allow_natives_syntax');
+
+const bench = common.createBenchmark(main, {
+  method: 'forEach iterator'.split(' '),
+  n: [1e6]
+});
+
+const str = 'one=single&two=first&three=first&two=2nd&three=2nd&three=3rd';
+
+function forEach(n) {
+  const params = new URLSearchParams(str);
+  const noop = () => {};
+  eval('%OptimizeFunctionOnNextCall(params.forEach)');
+  eval('%OptimizeFunctionOnNextCall(noop)');
```

Comment:
Look at https://github.com/nodejs/node/blob/master/benchmark/common.js#L211 .. there is a utility method that takes care of the V8 optimization bits.